File: src/main/java/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotificationEmail.java
package fi.metatavu.metaform.server.persistence.model.notifications;

import java.util.UUID;

import javax.persistence.Cacheable;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

/**
 * JPA entity representing email notification email
 * 
 * @author Antti Leppä
 */
@Entity
@Cacheable(true)
@Cache(usage = CacheConcurrencyStrategy.TRANSACTIONAL)
public class EmailNotificationEmail {

  @Id
  private UUID id;

  @ManyToOne(optional = false)
  private EmailNotification emailNotification;
  
  @Email
  @NotNull
  @NotEmpty
  private String email;
  
  public UUID getId() {
    return id;
  }
  
  public void setId(UUID id) {
    this.id = id;
  }
  
  public EmailNotification getEmailNotification() {
    return emailNotification;
  }
  
  public void setEmailNotification(EmailNotification emailNotification) {
    this.emailNotification = emailNotification;
  }
  
  public String getEmail() {
    return email;
  }
  
  public void setEmail(String email) {
    this.email = email;
  }
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/model/notifications/EmailNotificationEmail.kt
package fi.metatavu.metaform.server.persistence.model.notifications

import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import java.util.*
import jakarta.persistence.Cacheable
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull

/**
 * JPA entity representing email notification email
 *
 * @author Antti Leppä
 */
@Entity
@Cacheable(true)
@Cache(usage = CacheConcurrencyStrategy.TRANSACTIONAL)
class EmailNotificationEmail {
  @Id
  var id: UUID? = null

  @ManyToOne(optional = false)
  var emailNotification: EmailNotification? = null

  @Column
  @NotNull
  @NotEmpty
  @Email
  lateinit var email: String
}